added node runtime migration 10.x to latest (#3340)
aws-amplify/docs: docs/cli/migration/lambda-node-version-update.md

### runtime string replacement

With the latest version of the Amplify CLI (> 4.7.0), when you execute any `amplify` command on a project initialized by CLI version prior to 4.7.0, it will prompt for your confirmation to automatically upgrade your NodeJS Lambda runtime versions, from `nodejs8.10` to `nodejs10.x` in all the CloudFormation template files under the `amplify/backend` folder. If you do not confirm, you will need to manually carry out such replacements. You can go to each category subdirectory, then each resource subdirectory under it, and locate the template file (it could be either `.yml` or `.json` file), the template file has `template` in its name. Then do a global string replacement of `nodejs8.10` to `nodejs10.x` in the file.

## Node.js 10.x to Node.js latest version supported by AWS
According to AWS Lambda [Runtime Support Policy](https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html), AWS Lambda deprecates Node.js runtime Node.js 10.x on August 30th, 2021.

> Amplify CLI has been updated to replace Node.js 10.x with Node.js latest version for all files that it creates for you. Migration will be done automatically by running `amplify push` command, this will prompt users to migrate from Node.js 10.x to latest Node.js runtimes as supported by AWS. List of changes are summarized below:

Reviewer: ```suggestion
Amplify CLI has been updated to replace Node.js 10.x with the latest supported Node.js version for all resources that it creates for you. Migration will be done automatically by running the `amplify push` command, which will prompt to confirm the migration. List of changes are summarized below:
```

### runtime string replacement

With the latest version of the Amplify CLI (> 4.7.0), when you execute any `amplify` command on a project initialized by CLI version prior to 4.7.0, it will prompt for your confirmation to automatically upgrade your NodeJS Lambda runtime versions, from `nodejs8.10` to `nodejs10.x` in all the CloudFormation template files under the `amplify/backend` folder. If you do not confirm, you will need to manually carry out such replacements. You can go to each category subdirectory, then each resource subdirectory under it, and locate the template file (it could be either `.yml` or `.json` file), the template file has `template` in its name. Then do a global string replacement of `nodejs8.10` to `nodejs10.x` in the file.

## Node.js 10.x to Node.js 12.x
According to AWS Lambda [Runtime Support Policy](https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html), AWS Lambda deprecates Node.js runtime Node.js 10.x on August 30th, 2021.

> The Amplify CLI code base has been updated to reflect this change. Amplify CLI replaces Node.js 10.x with Node.js 14.x in the Lambda functions that it creates for you. Migration will be done automatically by running `amplify push` command, this will prompt users to migrate from Node.js 10.x to latest Node.js runtimes as supported by AWS.

Reviewer: Title says 10.x to 12.x, but the paragraph says the migration is from 10.x to 14.x - is this intentional?

Reviewer 2: Ping @akshbhu

Author: I have listed what all changes we will be doing automatically in node migration during amplify push
